File: CinemaBookingApp/CinemaBookingApp/CinemaBookingApp/Models/DTOs/Combo.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace CinemaBookingApp.Models.DTOs
{
    public class Combo
    {
        public class ComboDTO
        {
            public long ComboId { get; set; }

            public string Name { get; set; }

            public string ImageURL { get; set; }

            public int Quantity { get; set; }

            public double Price { get; set; }
        }

        public class ComboInTicketDTO
        {
            public long ComboId { get; set; }

            public int Quantity { get; set; }

            public string Name { get; set; }
        }

        public class UnAvailableComboDTO
        {
            public long ComboId { get; set; }

            public string Name { get; set; }
        }
    }
}

File: CinemaBookingApp/Front-end/CinemaBookingApp/src/types/combo.ts
import {SeatSelectionParamProps} from './seat';

export interface ComboProps {
  comboId: number;
  name: string;
  imageURL: string;
  quantity: number;
  price: number;
}

export interface SelectedComboProps {
  combo: ComboProps;
  quantity: number;
}

export interface ComboParamProps {
  seatParam: SeatSelectionParamProps;
  selectedCombos: SelectedComboProps[];
  totalPriceCombos: number;
}

export interface ComboInTicketProps {
  comboId: number;
  quantity: number;
  name: string;
}

export interface UnAvailableCombosProps {
  comboId: number;
  name: string;
}
